bridge/slack/slack.go

```diff
@@ -315,6 +321,39 @@ func (b *Bslack) sendRTM(msg config.Message) (string, error) {
 	return b.postMessage(&msg, messageParameters, channelInfo)
 }
 
+func (b *Bslack) updateTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (handled bool, err error) {
+	if msg.Event != config.EVENT_TOPIC_CHANGE {
+		return false, nil
+	}
+
+	if !b.GetBool("SyncTopicChange") {
+		return false, nil
+	}
+
+	incomingChangeType, text := b.extractTopicOrPurpose(msg.Text)
+	switch incomingChangeType {
+	case "topic":
+		if strings.HasSuffix(channelInfo.Topic.Value, "[nosync]") {
+			break
+		}
+		_, err = b.rtm.SetTopicOfConversation(channelInfo.ID, text)
+	case "purpose":
+		if strings.HasSuffix(channelInfo.Purpose.Value, "[nosync]") {
+			break
+		}
+		_, err = b.rtm.SetPurposeOfConversation(channelInfo.ID, text)
+	}
+
+	if err != nil {
```

Comment:
This logic is no longer necessary. If we reach this point (assuming my suggested changes above) we should log an error (as it should never happen). This best actually encoded in the `switch` itself by adding a `default:` case at the bottom. That way we don't need to specify a return after the `switch statement` as all cases are covered.

Reply:
Why did we create duplication here? What is inappropriate about doing the repetitive part outside the switch?

Reply:
I should have written the code in an editor, not in GitHub... indeed quite convoluted. However the original version was not good as it did not deal with rate-limiting correctly. Let me use my editor instead and post the suggestion here.

Reply:
So here the suggestion for the whole function. Sorry for the hiccup:
```go
func (b *Bslack) updateTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (bool, error) {
	if msg.Event != config.EVENT_TOPIC_CHANGE || !b.GetBool("SyncTopic") {
		return false, nil
	}

	incomingChangeType, text := b.extractTopicOrPurpose(msg.Text)

	var updateFunc func(channelID string, value string) (*slack.Channel, error)
	switch incomingChangeType {
	case "topic":
		updateFunc = b.rtm.SetTopicOfConversation
	case "purpose":
		updateFunc = b.rtm.SetPurposeOfConversation
	default:
		return true, fmt.Errorf("encountered unrecognised channel meta operation: %s", incomingChangeType)
	}

	for {
		_, err = updateFunc(channelInfo.ID, text)
		if err == nil {
			return true, nil
		}
		if err = b.handleRateLimit(err); err != nil {
			return true, err
		}
	}
}
```